test/integration/cases/07-sni-route/verify.sh

```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
```

Comment:
Please also add test cases verifying that:
- https requests to example.com and example.net present appropriate certificates - [stackoverflow](https://stackoverflow.com/a/7886248) - hopefully our test environment has openssl installed
- https requests to example.com/bar and example.net/foo yield 404
- http requests to example.com/foo and example.net/bar yield 404

Reply:
Added 404 check. The router doesn't actually handle choosing what certificate to present, and will happily SNI route even if the certificate doesn't actually match. You wouldn't do that in practice (the test YAML seeks to present a realistic configuration even though there are parts of it we don't care about), but as far as the proxy implementation is concerned, they're not actually coupled together.

Given that presenting the correct certificate isn't actually in scope for what the route SNI property does and that there's no good way to extract the certificate hostname (both curl and s_client will only present that in human-readable verbose output--neither has a dedicated "return CN only" tool), I think the 404 test should be sufficient here.

Reply:
Ah, I see. I have just realized that the current test (in this PR) does not verify the functionality changed by this PR.

We need a test that ensures that, if there is an `Ingress` resource with a `konghq.com/snis` annotation, then it won't match if a request comes with a different SNI. I wrote 2ecda3d3e3d24640ed11098d4b606d605530992f to fix that. Lines 14-15 fail for KIC without this PR, but pass for KIC with this PR. (This commit also adds further 404 cases that I asked for in a previous comment in this thread).